src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final double arrayVolumeAdd = 1.5;
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
 
+        if (size == elementData.length) {
+            grow(size);
+        }
+        elementData[size++] = value;
+
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == elementData.length) {
+            grow(size);
+        }
+        checkIndex(index, 0, size);
```

Comment:
I'd move validation to 1st place in order not to perform any actions if we have invalid input
```suggestion
        checkIndex(index, 0, size);
        if (size == elementData.length) {
            grow(size);
        }
```